telegram/_utils/enum.py

```diff
@@ -41,12 +45,49 @@ def get_member(enum: Type[_Enum], value: _A, default: _B) -> Union[_Enum, _A, _B
         return default
 
 
-class StringEnum(str, Enum):
-    """Helper class for string enums where the value is not important to be displayed on
+# Python 3.11 and above has a different output for mixin classes for IntEnum, StrEnum and IntFlag
+# see https://docs.python.org/3.11/library/enum.html#notes. We want e.g. str(StrEnumTest.FOO) to
+# return "foo" instead of "StrEnumTest.FOO", which is not the case < py3.11
+if sys.version_info < (3, 11):
+
+    class StringEnum(str, Enum):
+        """Helper class for string enums where the value is not important to be displayed on
+        stringification.
+        """
+
+        __slots__ = ()
+
+        def __repr__(self) -> str:
+            return f"<{self.__class__.__name__}.{self.name}>"
+
+        def __str__(self) -> str:
+            return str.__str__(self)
+
+else:
+
+    class StringEnum(StrEnum):
```

Comment:
does the if-else actually save something here? feels a bit like just declaring `StringEnum` manually would be enough, as we have to modify the buit-in `StrEnum` anyways. Once 3.10 is dropped, that changes, ofc